client/internal/httpsender_test.go

```diff
@@ -62,6 +63,49 @@ func TestHTTPSenderRetryForStatusTooManyRequests(t *testing.T) {
 	srv.Close()
 }
 
+func TestHTTPSenderRetryForStatusServiceUnavailable(t *testing.T) {
```

Comment:
Does this test case have anything to do with the new functionality added to Close() the body? It seems to verify unrelated functionality. If that is the case let's not add in the same PR. We want the PR to be about one subject - making Close() happen and having tests that verify that Close() happens.